apps/indexer/lib/indexer/fetcher/celo_voter_votes.ex

```diff
@@ -0,0 +1,133 @@
+defmodule Indexer.Fetcher.CeloVoterVotes do
+  @moduledoc """
+  Fetches Celo voter votes.
+  """
+  use Indexer.Fetcher
+  use Spandex.Decorators
+
+  require Logger
+
+  alias Explorer.Celo.{AccountReader, ContractEvents}
+  alias Explorer.Chain
+  alias Explorer.Chain.{Block, Hash}
+  alias Explorer.Chain.CeloVoterVotes, as: CeloVoterVotesChain
+
+  alias ContractEvents.Election.ValidatorGroupVoteActivatedEvent
+
+  alias Indexer.BufferedTask
+  alias Indexer.Fetcher.Util
+
+  @behaviour BufferedTask
+
+  @spec async_fetch([%{block_hash: Hash.Full.t(), block_number: Block.block_number()}]) :: :ok
+  def async_fetch(blocks) when is_list(blocks) do
+    filtered_blocks =
+      blocks
+      |> Enum.filter(&(rem(&1.block_number, 17280) == 0))
+
+    BufferedTask.buffer(__MODULE__, filtered_blocks)
+  end
+
+  @doc false
+  def child_spec([init_options, gen_server_options]) do
+    init_options_with_polling =
+      init_options
+      |> Keyword.put(:poll, true)
+      |> Keyword.put(:poll_interval, :timer.minutes(60))
+      |> Keyword.put(:max_concurrency, 1)
+      |> Keyword.put(:max_batch_size, 1)
+
+    Util.default_child_spec(init_options_with_polling, gen_server_options, __MODULE__)
+  end
+
+  @impl BufferedTask
+  def init(initial, reducer, _json_rpc_named_arguments) do
+    {:ok, final} =
+      Chain.stream_blocks_with_unfetched_voter_votes(initial, fn block, acc ->
+        block
+        |> reducer.(acc)
+      end)
+
+    final
+  end
+
+  @impl BufferedTask
+  def run(entries, _json_rpc_named_arguments) do
+    failed_list =
+      entries
+      |> Enum.map(&get_previous_epoch_voters_and_groups/1)
+      |> List.flatten()
+      |> Enum.uniq_by(&{&1.block_hash, &1.account_hash, &1.group_hash})
+      |> Enum.map(&fetch_from_blockchain/1)
+      |> import_items()
+
+    if failed_list == [] do
+      :ok
+    else
+      {:retry, failed_list}
+    end
+  end
+
+  def get_previous_epoch_voters_and_groups(%{block_hash: block_hash, block_number: block_number}) do
+    activated_in_previous_epoch = ValidatorGroupVoteActivatedEvent.voters_activated_votes_in_last_epoch(block_number)
```

Comment:
We could calculate the previous epoch block number here to avoid repetition and pass it to those two functions (renaming them to something more appropriate like `voters_activated_votes_in_epoch` and `non_zero_voter_votes_in_epoch`).